src/assignment5/DAG.java

```diff
@@ -0,0 +1,102 @@
+package assignment5;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Directed graph having a node for each character.
+ * 
+ */
+public class DAG {

```

Comment:
Does not have to be public - it is an implementation detail of Assignment5 and likely not meant for use in other places.

If the class is meant to be public, its name should probably be change to something that more closely describes what it does - current name suggests a general purpose DAG implementation, while in fact it is a very specialized application of DAG that has only one specific use.